playbooks/roles/infrastructure/tasks/aws/verifications.yml

```diff
@@ -0,0 +1,101 @@
+---
+
+# If vpc_id is given, public_subnet_ids and private_subnet_ids are required
+- block:
+  - name: Check if the user-defined vpc_id exists
+    ec2_vpc_net_info:
+      vpc_ids: "{{ vpc_id }}"
+  
+  - name: Fail when subnet_ids are undefined
+    fail:
+      msg: |
+        ERROR: When vpc_id is provided a list of public and private subnet_ids
+        must also be provided.  For example:
+          public_subnet_ids:
+            - subnet-0123456789abcdef0
+          private_subnet_ids:
+            - subnet-1234567890abcdef1
+            - subnet-234567890abcdef12
+            - subnet-34567890abcdef123
+    when: (public_subnet_ids is undefined) or (private_subnet_ids is undefined)
```

Comment:
We'll need a bit more complex check because for a disconnected environment, there will not be any `public_subnet_ids`. Since we don't have the full logic for disconnected yet, this is fine to leave as is. Just commenting so we have a reminder for the future.